projects/storefrontapp-e2e-cypress/cypress/helpers/b2b/my-company/my-company.utils.ts

```diff
@@ -0,0 +1,154 @@
+import { MyCompanyRowConfig, MyCompanyConfig } from './models/index';
+import { testListFromConfig } from './my-company-list';
+import { testCreateUpdateFromConfig } from './my-company-form';
+import { nextPage } from '../../product-search';
+import { POWERTOOLS_BASESITE } from '../../../sample-data/b2b-checkout';
+
+export function testMyCompanyFeatureFromConfig(config: MyCompanyConfig) {
+  describe(`My Company - ${config.name}`, () => {
+    before(() => {
+      Cypress.env('BASE_SITE', POWERTOOLS_BASESITE);
+    });
+
+    testListFromConfig(config);
+    testCreateUpdateFromConfig(config);
+  });
+}
+
+/**
+ * Assert table headers are correctly labelled.
+ * @param configs Row configurations containing header labels.
+ */
+export function checkRowHeaders(configs: MyCompanyRowConfig[]): void {
+  configs.forEach((config: any) => {
+    if (config.showInTable) {
+      cy.get('th').should('contain.text', config.label);
+    }
+  });
+}
+
+export function checkRows(rows): void {
+  let j = 1; // Skip header table row at 0
+  rows.forEach((row: any) => {
+    if (row.text.length) {
+      cy.get('tr')
+        .eq(j)
+        .within(() => {
+          for (let i = 0; i < row.text.length; i++) {
+            if (row.text[i]) {
+              if (Array.isArray(row.text[i])) {
+                // Used in user roles array
+                // Because we can't use translate pipe, have to check per case
+                row.text[i].forEach((text) => {
+                  switch (text) {
+                    case 'b2bcustomergroup':
+                      return cy.get('td').eq(i).contains('Customer');
+                    case 'b2bmanagergroup':
+                      return cy.get('td').eq(i).contains('Manager');
+                    case 'b2bapprovergroup':
+                      return cy.get('td').eq(i).contains('Approver');
+                    case 'b2badmingroup':
+                      return cy.get('td').eq(i).contains('Admin');
```

Comment:
Not really convinced if we should keep that kind of exception in generic code. Maybe additional config rows could do the job?

Reply:
I think this kind of improvement can be done in the users PR or as a stretch since this pr doesn't have the users config to test against. Its a little complicated to be elegant since we have to read the variable in raw format and translate it.